src/ray/raylet/node_manager.cc

```diff
@@ -2440,6 +2442,48 @@ void NodeManager::HandleGetNodeStats(const rpc::GetNodeStatsRequest &node_stats_
   }
 }
 
+void NodeManager::HandleGetResourceUsageByTask(
+    const rpc::GetResourceUsageByTaskRequest &request,
+    rpc::GetResourceUsageByTaskReply *reply,
+    rpc::SendReplyCallback send_reply_callback) {
+  RAY_LOG(INFO) << "HandleGetResourceUsageByTask: lease workers: "
+                << leased_workers_.size();
+  auto total_map = reply->mutable_total();
+  auto available_map = reply->mutable_available();
+  auto node_resources =
+      cluster_resource_scheduler_->GetLocalResourceManager().GetLocalResources();
+  for (const auto &resource : node_resources.total.ToResourceRequest().ToResourceMap()) {
+    (*total_map)[resource.first] = resource.second;
+  }
+  for (const auto &resource :
+       node_resources.available.ToResourceRequest().ToResourceMap()) {
+    (*available_map)[resource.first] = resource.second;
+  }
+  // We get the logical resources acquired by each worker that the raylet has leased out
+  RAY_LOG(INFO) << "HandleGetResourceUsageByTask: lease workers: "
+                << leased_workers_.size();
+  for (const auto &worker_it : worker_pool_.GetAllRegisteredWorkers(
+           /*filter_dead_workers*/ true, /*filter_inactive_workers*/ true)) {
+    // Check that worker is also leased
```

Comment:
This can have check failure if worker is blocked. So let's remove the check!

Reply:
yes, that's the point! leased_workers_ should be a superset of the "active" workers

Reply:
Yes, that's the point! Leased workers is supposed to be a superset of "active" workers! (it is the union of blocked workers and active workers)